test/components/map.spec.js

```diff
@@ -38,38 +100,65 @@ test('InteractiveMap#named export', t => {
   t.end();
 });
 
-test('InteractiveMap#call onLoad when provided', t => {
-  function onLoad(...args) {
-    t.is(args.length, 0, 'onLoad does not expose the map object.');
-    t.end();
-  }
+TEST_CASES.forEach(testCase => {
+  test(`InteractiveMap#load ${testCase.title}`, t => {
+    let result = null;
+    let timer = null;
 
-  const props = Object.assign({}, defaultProps, {onLoad});
+    function end() {
+      const warning = result.root.findAllByType('h3');
+      if (testCase.shouldLoad) {
+        t.notOk(warning.length, 'shouldn\'t show warning');
+      } else {
+        t.ok(warning.length, 'should show warning');
+      }
 
-  const map = createElement(InteractiveMap, props);
+      result.unmount();
+      t.end();
+    }
 
-  const result = ReactTestRenderer.create(map);
+    function onLoad() {
+      if (testCase.shouldLoad) {
+        t.pass('onLoad is called');
+      } else {
+        t.fail('onLoad should not be called');
+      }
+      clearTimeout(timer);
+      end();
+    }
 
-  t.ok(result, 'InteractiveMap rendered');
+    const props = Object.assign({}, testCase.props, {onLoad});
 
-  if (!InteractiveMap.supported()) {
-    t.ok('onLoad not called since InteractiveMap.supported() false');
-    t.end();
-  } else {
-    /* global setTimeout */
-    setTimeout(() => {
-      t.fail('onLoad wasn\'t called');
+    const map = createElement(InteractiveMap, props);
+
+    result = ReactTestRenderer.create(map);
+
+    t.ok(result, 'InteractiveMap rendered');
+
+    if (!InteractiveMap.supported()) {
+      t.ok('onLoad not called since InteractiveMap.supported() false');
+      result.unmount();
       t.end();
-    }, 5000);
-  }
+    } else {
+      timer = setTimeout(() => {
+        if (testCase.shouldLoad) {
+          t.fail('onLoad wasn\'t called');
+        } else {
+          t.pass('onLoad wasn\'t called');
+        }
+        end();
+      }, 5000);
```

Comment:
This is a pretty long timeout, can be worth checking how much impact it has on test suite execution time after all your changes settle. 

Is it still only called once or now for every TC?

Reply:
When `onLoad` is called, this timer is cleared and the test ends. If all test cases run correctly, only one of them will actually hit the time out.